server/src/graphql/schema.graphql

```diff
@@ -11,81 +11,98 @@
 schema {
   query: Query
   mutation: Mutation
-  subscription: Subscription
+  # subscription: Subscription
 }
 
+# type Subscription {
+#   surveyUpdates (surveyId: Int!): Survey
+# }
+
 type Query {
-  # Returns the logged-in user making the query, or null if anonymous.
-  self: User
+  # Returns a Party if one exists with the partyName, and its optional password matches the optional partyPassword.
+  # Otherwise, null.
+  party (partyName: String!, partyPassword: String): Party
 
-  surveys: [Survey!]!
-  survey (surveyId: Int!): Survey
+  # Returns all Song sorted by song title.
+  songs: [Song!]!
 }
 
 type Mutation {
-  # Records an answer to a survey question presented in class.
-  answerSurvey (input: SurveyInput!): Boolean!
-
-  # Moves the survey to the next question (or starts it if it hasn't started). ADMIN only.
-  nextSurveyQuestion (surveyId: Int!): Survey
-}
-
-type Subscription {
-  surveyUpdates (surveyId: Int!): Survey
+  # Creates a Party with the optional partyPassword if one does not yet exist with the partyName.
+  # Returns that newly created Party, or null if a Party already exists with the partyName.
+  createParty (partyName: String!, partyPassword: String): Party
+
+  # Votes for a Song in the context of a Party.
+  # Returns the VotedSong if the voting was successful (partyId and songId must exist), or null if unsuccessful.
+  vote (partyId: Int!, songId: Int!): VotedSong
+
+  # Advances the CurrentSong of a Party to the next song, which is the VotedSong with the highest VotedSong.count
+  # out of all the VotedSong for the Party. If there is no VotedSong for the Party, the CurrentSong becomes null.
+  # Returns the Party if successful, or null if unsuccessful.
+  nextSong(partyId: Int!): Party
```

Comment:
Should we add another comment specifying that the skipped song will be added to `playedSongs`

Reply:
Sure, I'll do that once we agree on all the changes to make.